collection.go

```diff
@@ -33,6 +33,15 @@ type CollectionOptions struct {
 	// The given Maps are Clone()d before being used in the Collection, so the
 	// caller can Close() them freely when they are no longer needed.
 	MapReplacements map[string]*Map
+
+	// UnsafeVariableExperiment enables the unsafe variable experiment, allowing
+	// the use of [VariablePointer] for direct memory access and atomic operations
+	// on global bpf variables.
+	//
+	// Experimental: enabling this may cause segfaults or compromise the memory
```

Comment:
We should also document that this may go away at any point in time if it is necessary for the health of the ecosystem. Go runtime >> this feature.

Reply:
The goal is to never let things get to this point. The path has been charted using `AddForeignCleanup`, so if this workaround ends up painting the runtime into a corner, that's all the more reason to get foreign cleanup shipped ASAP.